spec/controllers/companies_controller_spec.rb

```diff
@@ -0,0 +1,168 @@
+require "rails_helper"
+include SessionsHelper
+
+describe CompaniesController, type: :controller do
+  let!(:account_company){FactoryBot.create :account, role: "company"}
+  let!(:account_company2){FactoryBot.create :account, role: "company"}
+  let!(:account_user){FactoryBot.create :account, role: "user"}
+  let!(:company){FactoryBot.create :company, account_id: account_company.id}
+  let!(:user){FactoryBot.create :user, account_id: account_user.id}
+
+  before{log_in account_company}
```

Comment:
before này chạy ntn

Reply:
before là cách viết ngắn gọn của before(:each), các câu lệnh trong before sẽ được chạy trước mỗi example (khối it) trong phạm vi nó đứng. Ở đây log_in account_company sẽ đăng nhập tài khoản công ty trước khi chạy tất cả các example trong file test này